File: sec1/python/heron.py
def heron_sqrt(n):
    guess = n
    prev_guess = 0
    while round(guess*1000)/1000 != round(prev_guess*1000)/1000:
        prev_guess = guess
        guess = .5*(guess + n/guess)
    return guess

print(heron_sqrt(50))

File: sec1/js/heron.js
function heronSqrt(n) {
    var guess = n, prevGuess = 0;
    while (guess.toFixed(3) != prevGuess.toFixed(3)) {
        prevGuess = guess;
        guess = .5*(guess + n/guess);
    }
    return guess;
}

console.log(heronSqrt(50));